src/networking.c

```diff
@@ -2437,6 +2437,13 @@ int processPendingCommandsAndResetClient(client *c) {
         c->flags &= ~CLIENT_PENDING_COMMAND;
         if (processCommandAndResetClient(c) == C_ERR) {
             return C_ERR;
+        } else if (c->flags & CLIENT_MASTER && c->repl_applied) {
+            /* The pending command is applied, we can propagate the data
+             * and trim it from querybuf now. */
+            replicationFeedStreamFromMasterStream(c->querybuf,c->repl_applied);
```

Comment:
what was the reason to move this from `commandProcessed` to here?
i think that logically `replicationFeedStreamFromMasterStream` should be done from `commandProcessed`.

also, aren't we still doing `sdsrange` on each command we process?
correct me if i'm wrong, but i theory we could have kept feeding the replication stream per command, but just avoid the sdsrange.
i.e. remember to which offset we already fed and leave the sdsrange for the end of processInputBuffer.

Reply:
about `processPendingCommandsAndResetClient`, please pay attention to the key word `Pending`, it's a very special flag means the command is not applied yet, there are two scenarios:
1. master client blocked(by clients pause), the current command is processed but not applied, and now we reprocess it.
2. the IO threads are active in read, and master client's current is parsed in IO thread but not processed, and now we process it.

so we must feed replication stream here

Reply:
ohh, sorry, i didn't realize i'm looking at `processPendingCommandsAndResetClient` and it's use case.
for some reason i thought that's it's somewhere in the flow of `processInputBuffer`.
still, i would prefer to do the feeding from `commandProcessed` (i think that it's more logically fitting the role of that function), and the trimming in `processInputBuffer`.
if we do that, maybe we don't need any trimming in `processPendingCommandsAndResetClient`? (i.e. that'll happen on the next time we'll get to `processInputBuffer`, which is immediately after).

Reply:
suggestion accepted :)